ConfigSpace/io/pcs_new.py

```diff
@@ -0,0 +1,475 @@
+        line = line.strip()
+        if "|" in line:
+            # It's a condition
+            try:
+                c = pp_condition.parseString(line)
+                conditions.append(c)
+            except pyparsing.ParseException:
+                raise NotImplementedError("Could not parse condition: %s" % line)
+            continue
+        if "}" not in line and "]" not in line:
+            continue
+        if line.startswith("{") and line.endswith("}"):
+            forbidden.append(line)
+            continue
+        if len(line.strip()) == 0:
+            continue
+
+        ct += 1
+        param = None
+
+        create = {"int": UniformIntegerHyperparameter,
+                  "float": UniformFloatHyperparameter,
+                  "categorical": CategoricalHyperparameter,
+                  "ordinal": OrdinalHyperparameter}
+
+        try:
+            param_list = pp_cont_param.parseString(line)
+            log = param_list[10:]
+            if len(log) > 0:
+                log = log[0]
+            param_list = param_list[:10]
+            name = param_list[0]
+            paramtype = "int" if "integer" in param_list else "float"
+            lower = float(param_list[3])
+            upper = float(param_list[5])
+            log_on = True if "log" in log else False
+            default = float(param_list[8])
+            param = create[paramtype](name=name, lower=lower, upper=upper,
+                                      q=None, log=log_on, default=default)
+            cont_ct += 1
+        except pyparsing.ParseException:
+            pass
+
+        try:
+            if "categorical" in line:
+                param_list = pp_cat_param.parseString(line)
+                name = param_list[0]
+                choices = [choice for choice in param_list[3:-4:2]]
+                default = param_list[-2]
+                param = create["categorical"](name=name, choices=choices, default=default)
+                cat_ct += 1
+                
+            elif "ordinal" in line:
+                param_list = pp_ord_param.parseString(line)
+                name = param_list[0]
+                sequence = [seq for seq in param_list[3:-4:2]]
+                default = param_list[-2]
+                param = create["ordinal"](name=name, sequence=sequence, default=default)
+                ord_ct += 1
+                
+        except pyparsing.ParseException:
+            pass
+
+        if param is None:
+            raise NotImplementedError("Could not parse: %s" % line)
+
+        configuration_space.add_hyperparameter(param)
+
+    for clause in forbidden:
+        param_list = pp_forbidden_clause.parseString(clause)
+        tmp_list = []
+        clause_list = []
+        for value in param_list[1:]:
+            if len(tmp_list) < 3:
+                tmp_list.append(value)
+            else:
+                # So far, only equals is supported by SMAC
+                legal = ['=', '>', '<']
```

Comment:
The two lines above contradict each other. According to the format definition I got from Marius, > and < aren't part of this new pcs format.